Assets/MixedRealityToolkit/MSBuild/Scripts/CSProjectInfo.cs

```diff
@@ -0,0 +1,363 @@
+
+            InEditorPlatforms = GetCompilationPlatforms(true);
+            PlayerPlatforms = GetCompilationPlatforms(false);
+
+            if (InEditorPlatforms.Count == 0 && PlayerPlatforms.Count == 0)
+            {
+                Debug.LogError($"The assembly project '{Name}' doesn't contain any supported in-editor or player platform targets.");
+            }
+        }
+
+        private ProjectType GetProjectType(AssemblyDefinitionInfo assemblyDefinitionInfo, Assembly assembly)
+        {
+            if (assemblyDefinitionInfo != null)
+            {
+                return assemblyDefinitionInfo.EditorPlatformSupported && !assemblyDefinitionInfo.NonEditorPlatformSupported ? ProjectType.EditorAsmDef : ProjectType.AsmDef;
+            }
+
+            switch (assembly.name)
+            {
+                case "Assembly-CSharp":
+                case "Assembly-CSharp-firstpass":
+                    return ProjectType.PredefinedAssembly;
+                case "Assembly-CSharp-editor":
+                case "Assembly-CSharp-firstpass-editor":
+                    return ProjectType.PredefinedEditorAssembly;
+                default:
+                    Debug.LogError($"Predefined assembly '{assembly.name}' was not recognized, this generally means it should be added to the switch statement in CSProjectInfo:GetProjectType. Treating is as a PredefinedAssembly instead of PredefinedEditorAssembly.");
+                    return ProjectType.PredefinedAssembly;
+            }
+        }
+
+        private ReadOnlyDictionary<BuildTarget, CompilationPlatformInfo> GetCompilationPlatforms(bool inEditor)
+        {
+            bool returnAllPlatforms = ProjectType == ProjectType.PredefinedAssembly
+                || (inEditor && ProjectType == ProjectType.PredefinedEditorAssembly)
+                || (inEditor && ProjectType == ProjectType.EditorAsmDef)
+                || (inEditor && ProjectType == ProjectType.AsmDef && AssemblyDefinitionInfo.EditorPlatformSupported);
+
+            if (returnAllPlatforms)
+            {
+                return new ReadOnlyDictionary<BuildTarget, CompilationPlatformInfo>(availablePlatforms.ToDictionary(t => t.BuildTarget, t => t));
+            }
+
+            bool returnNoPlatforms = (!inEditor && ProjectType == ProjectType.PredefinedEditorAssembly)
+                || (!inEditor && ProjectType == ProjectType.EditorAsmDef)
+                || (!inEditor && ProjectType == ProjectType.AsmDef && AssemblyDefinitionInfo.TestAssembly);
+
+            if (returnNoPlatforms)
+            {
+                return new ReadOnlyDictionary<BuildTarget, CompilationPlatformInfo>(new Dictionary<BuildTarget, CompilationPlatformInfo>());
+            }
+
+            // We only are an asmdef at this point, as we handle all predefined assembly at line 130, inEditor && predefined editor assembly at line 130, and !inEditor && predefined editor assembly at line 139
```

Comment:
>line 130, inEditor && predefined editor assembly at line 130, and !inEditor && predefined editor assembly at line 139 [](start = 92, length = 117)

Nit: I would probably add some other named comment sections:

i.e. on line 130, have a comment saying "predefined assembly reconciliaion section", and then here refer to the named section.

Line numbers can and will change arbitrarily and this comment will go out of date pretty easily (and it's really easy to forget updating this as you change any of the above lines)